pre-commit

```diff
@@ -47,3 +47,15 @@ error() {
 cargo clippy --tests -- -D warnings --D clippy::pedantic || error "Clippy errors"
 
 cargo test || error "Test failures"
+
+which flake518 || error "flake518 not found. Install with \`python3 -m pip install flake518\` in the desired python environment."
```

Comment:
These might do better with a venv instead.

Reply:
Yea, I thought about that. I figured leaving it up to the user to figure out if they want to use a venv, but we can also go that route.

Reply:
@martinthomson  any objection to merging like this, and setting that up if/when we need it? (trying to unblock while i'm on vacation)